bin/publish.ls

```diff
@@ -0,0 +1,150 @@
+require! {
+  'chalk' : {bold, red, cyan, yellow, green, dim}
+  'inquirer'
+  'jsonfile'
+  'semver'
+  'shelljs/global'
+}
+
+
+SUBPROJECTS_TO_PUBLISH =
+  'exocom-mock-javascript'
+  'exocom-server'
+  'exorelay-javascript'
+  'exoservice-javascript'
+
+# exocom-server is used because there is no root package.json
+# all node exocommunication services are in lock-step versioning
+{name, version} = jsonfile.readFileSync './exocom-server/package.json'
+
+if process.argv.length != 3
+  display-help!
+
+level = process.argv[2]
+display-help! if level in ['-h, --help']
+
+target-version = get-target-version version, level
+
+confirm-target-version ->
+  ensure-no-open-files!
+  ensure-on-master!
+
+  check-npm-dependencies!
+  build-subprojects!
+  run-tests!
+  bump-version-numbers!
+  push-version-numbers!
+  publish-to-npm!
+  push-exocom-docker-image!
+  push-git-tags!
+
+
+function check-npm-dependencies
+  console.log green "Checking npm dependencies...\n"
+  run-command-in-subdirs do
+    command: './node_modules/.bin/update-check'
+    command-message: 'Checking dependencies'
+  console.log!
+
+
+function build-subprojects
+  console.log green "Building subprojects...\n"
+  run-command-in-subdirs do
+    command: './bin/build'
+    command-message: 'Building'
+  console.log!
+
+
+function run-tests
+  console.log green "Running tests in subprojects...\n"
+  run-command-in-subdirs do
+    command: './bin/spec'
+    command-message: 'Running tests in'
```

Comment:
Yes, but combined with `"#{command-message} subproject #{cyan directory}"` in `run-command-in-subdirs` it should